### ISSUE TITLE
[UI] (2.3) - Azure and VMware node templates are tagged to incorrect cloud credentials

### ISSUE THREAD

**Issue by davidnuzik** [rancher/rancher#22662]
Version: v2.2.8 (tracked in https://github.com/rancher/rancher/issues/22663)
Version: master-head (v2.3) (9/5/19)

**What kind of request is this (question/bug/enhancement/feature request):**
Bug, major

**Steps to reproduce (least amount of steps as possible):**
Reference: https://github.com/rancher/rancher/issues/18737

1. In a Rancher HA set up (or single-node rancher), create an Amazon cloud credential.
2. Create an Azure/VMware cloud credential.
3. Go to 'Node Templates' page.
4. Create an Azure/VMware node template. 
5. Select the appropriate azure/vmware cloud credential while creating the template.
6. After creation of the template, click on 'View in API'
7. Once you have multiple cloud credentials to pick from (azure, vsphere, do) if you attempt to create a DO nodeTemplate, you'll get an error in the UI: 
![image](https://user-images.githubusercontent.com/45179589/64364754-71baff00-cfc8-11e9-94a3-d3f8b7c9031c.png)
This happens even though we have properly selected the "DO" cloudCredential in the UI.

**Expected Result:**
`cloudCredentialId` should be tagged to the azure/vmware cloud credential selected while creating the template.

**Actual Result:**
`cloudCredentialId` is tagged to Amazon cloud credential, which is the first one on the Cloud Credentials page.

**Comment by davidnuzik** [comment]
Here is an example of what we see. In the UI it appears to properly only show the relevant cloud credentials for the nodeTemplate you are trying to create. So, in our case we only have one azure, amazon, and vsphere cloud credential. If I try to create an azure nodeTemplate the amazon cloudcredential is used even though the UI picked azure for us.

![image](https://user-images.githubusercontent.com/45179589/64365593-5a304600-cfc9-11e9-8762-10ae648386ab.png)

It may send wrong config too.

The linked issue https://github.com/rancher/rancher/issues/18737 has more information and here was the past pr that had fixed the issue (which is now regressed): https://github.com/rancher/ui/pull/2738

**Comment by davidnuzik** [comment]
I am not always seeing the issue in v2.3 (master-head (9/5/19) so I am going to provision another host and try again to make sure this is reproducable in v2.3;

I do always see the issue in v2.2.8 though. Create vsphere nodeTemplate and amazon cc is used.

**Comment by soumyalj** [comment]
Observed the same issue of " Unable to authenticate you." in v2.2.8.
Header: 
```
:authority: abc..rancher.space
:method: GET
:path: /meta/proxy/api.digitalocean.com/v2/regions?per_page=100
:scheme: https
accept: application/json
accept-encoding: gzip, deflate, br
accept-language: en-US,en;q=0.9
cookie: CSRF=74981c7dbf; R_USERNAME=admin; R_SESS=token-6gtlg:5m9l6db6trd6x9wlqxxzvsm8kt7mtzl7bfdz8hmvn87pn4ml84744j
referer: https://soumyahatest.qa.rancher.space/n/node-templates
sec-fetch-mode: cors
sec-fetch-site: same-origin
user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.100 Safari/537.36
x-api-cattleauth-header: Bearer credID=cattle-global-data:awscred passwordField=accessToken

```
Response:
```
{"id": "unauthorized", "message": "Unable to authenticate you." }

```

**Comment by davidnuzik** [comment]
Version: master-head (v2.3) (9/6/19)

I tested this carefully, following the steps to reproduce and testing both azure and vsphere then other providers. They ALL send the proper cloudCredentialId now in the POST request body. The clusters properly come up when I try to leverage the nodeTemplates too.

I also ensured that I can create a DO cluster (I do not run into the "Unable to authenticate you." issue).

I didn't identify any issues with my testing. UI/HTTP/API look good.
